functions/Move-DbaDbFile.ps1

```diff
@@ -0,0 +1,438 @@
+
+    .PARAMETER FileStructureOnly
+        Outputs an hash table defintion with current database file structure.
+        You can use this to define de destination folder and delete the entries you want to exclude. Then you can use this hashtable on -FileToMove parameter
+
+    .PARAMETER Force
+        This translates to instantly rolling back any open transactions that may be stopping the process of setting database offline.
+
+    .PARAMETER WhatIf
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Database, Move, File
+        Author: Cláudio Silva (@claudioessilva), claudioeesilva.eu
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Move-DbaDbFile
+
+    .EXAMPLE
+        PS C:\> Move-DbaDbFile -SqlInstance sql2017 -Database dbatools -FileType Data -FileDestination "D:\DATA2"
+
+        Copy all data files of dbatools database on sql2017 instance to the "D:\DATA2" path.
+        Before it puts database offline and after copy each file will update database metadata and it ends by set the database back online
+
+    .EXAMPLE
+        PS C:\> $fileToMove=@{
+            'dbatools'='D:\DATA3'
+            'dbatools_log'='D:\LOG2'
+        }
+        PS C:\> Move-DbaDbFile -SqlInstance sql2019 -Database dbatools -FileToMove $fileToMove
+
+        Declares a hashtable that says for each logical file the new path.
+        Copy each dbatools database file referenced on the hashtable on the sql2019 instance from the current location to the new mentioned location (D:\DATA3 and D:\LOG2 paths).
+        Before it puts database offline and after copy each file will update database metadata and it ends by set the database back online
+
+    .EXAMPLE
+        PS C:\> Move-DbaDbFile -SqlInstance sql2017 -Database dbatools -FileStructureOnly
+
+        Shows the current database file structure (without filenames).
+
+        Example:
+            $fileToMove=@{
+                'dbatools'='D:\Data'
+                'dbatools_log'='D:\Log'
+            }
+
+    #>
+
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Medium")]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter]$SqlInstance,
+        [PSCredential]$SqlCredential,
```

Comment:
Since remoting is involved to the server for supporting moving files over `Invoke-Command`, and use of `Start-BitsTransfer` this requires the need for `Credential` parameter as well.

Both commands accept a parameter of `-Credential` and the use of the `SqlCredential` should not be used for those calls.